lib/commons/blocs/preferences_bloc.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+// ----- EVENTS -----
+
+class PreferencesEvent extends Equatable {
+  @override
+  List<Object> get props => [];
+}
+
+class PreferencesLoadEvent extends PreferencesEvent {}
+
+class PreferencesSaveEvent extends PreferencesEvent {
+  final bool automaticDarkTheme;
+  final bool darkTheme;
+
+  PreferencesSaveEvent({this.automaticDarkTheme, this.darkTheme});
+}
+
+// ----- STATES -----
+
+class PreferencesState extends Equatable {
+  @override
+  List<Object> get props => [];
+}
+
+class PreferencesUninitializedState extends PreferencesState {}
+
+class PreferencesLoadingState extends PreferencesState {}
+
+class PreferencesSavingState extends PreferencesState {}
+
+class PreferencesLoadedState extends PreferencesState {
+  final bool automaticDarkTheme;
```

Comment:
Is it automatic or automatic for dark?
```suggestion
  final bool useAutomaticTheme;
```